bootrom/bootrom.c

```diff
@@ -153,28 +153,35 @@ void UsbPacketReceived(uint8_t *packet, int len) {
         case CMD_FINISH_WRITE: {
             uint32_t *flash_mem = (uint32_t *)(&_flash_start);
             for (int j = 0; j < 2; j++) {
+                uint32_t flash_address = arg0 + (0x100 * j);
+                AT91PS_EFC efc_bank = AT91C_BASE_EFC0;
+                int offset = 0;
+                uint32_t page_n = (flash_address - ((uint32_t)flash_mem)) / AT91C_IFLASH_PAGE_SIZE;
+                if (page_n >= 1024) {
+                    page_n -= 1024;
+                    efc_bank = AT91C_BASE_EFC1;
+                    // We need to offset the writes or it will not fill the correct bank write buffer.
+                    offset = 1024 * AT91C_IFLASH_PAGE_SIZE / 4;
```

Comment:
where did you find that magic offset value? Is there another more explicit way to express it?

Reply:
ha ok I get it, it's to reconstruct the real address, I was puzzled by the /4 but that's because flash_mem is in dwords.
maybe write it as:
offset = (AT91C_IFLASH_NB_OF_PAGES / 2) * AT91C_IFLASH_PAGE_SIZE / sizeof(uint32_t);